In [2]:
import re
import pandas as pd
import matplotlib.pyplot as plt

In [3]:
apiKey = '0F68F5F96EC6CC04C40F0A28024E20C0'
accountID = '96887717'
id = 'https://api.steampowered.com/IDOTA2Match_570/GetMatchHistory/V001/?key='+ apiKey\
+'&matches_requested=100&account_id='+ accountID+'&min_players=10'


In [12]:
matches = set()
#129 because the highest hero ID is 129
for i in range (129):
    print('Len of matches: ' + str(len(matches))+ ' before itteration ' + str(i))
    id = 'https://api.steampowered.com/IDOTA2Match_570/GetMatchHistory/V001/?key='+ apiKey\
    +'&hero_id=' + str(i)\
    +'&game_mode=1'\
    +'&matches_requested=100&account_id='+ accountID\
    +'&min_players=10'
    try:
        #errors out when it gets to a character I haven't played
        table = pd.DataFrame(pd.read_json(id).loc['matches']['result'])
        match_list = table['match_id'].to_list()
    except: 
        print('ERROR WITH ID: ' + str(i))
    print('   Length of matches with ID: '+str(i)+' is : '+str(len(match_list)))
    for match in match_list:
        matches.add(match)

Len of matches: 0 at itteration 0
Length of matches whith ID: 0 is : 100
Len of matches: 100 at itteration 1
Length of matches whith ID: 1 is : 28
Len of matches: 128 at itteration 2
Length of matches whith ID: 2 is : 40
Len of matches: 168 at itteration 3
Length of matches whith ID: 3 is : 10
Len of matches: 178 at itteration 4
Length of matches whith ID: 4 is : 29
Len of matches: 207 at itteration 5
Length of matches whith ID: 5 is : 25
Len of matches: 232 at itteration 6
Length of matches whith ID: 6 is : 17
Len of matches: 249 at itteration 7
Length of matches whith ID: 7 is : 31
Len of matches: 279 at itteration 8
Length of matches whith ID: 8 is : 21
Len of matches: 300 at itteration 9
Length of matches whith ID: 9 is : 13
Len of matches: 313 at itteration 10
Length of matches whith ID: 10 is : 9
Len of matches: 322 at itteration 11
Length of matches whith ID: 11 is : 18
Len of matches: 340 at itteration 12
Length of matches whith ID: 12 is : 12
Len of matches: 352 at itteration 

Length of matches whith ID: 107 is : 10
Len of matches: 2467 at itteration 108
Length of matches whith ID: 108 is : 47
Len of matches: 2514 at itteration 109
Length of matches whith ID: 109 is : 14
Len of matches: 2528 at itteration 110
Length of matches whith ID: 110 is : 8
Len of matches: 2533 at itteration 111
Length of matches whith ID: 111 is : 37
Len of matches: 2569 at itteration 112
Length of matches whith ID: 112 is : 10
Len of matches: 2578 at itteration 113
Length of matches whith ID: 113 is : 7
Len of matches: 2585 at itteration 114
Length of matches whith ID: 114 is : 15
Len of matches: 2600 at itteration 115
Length of matches whith ID: 115 is : 100
Len of matches: 2600 at itteration 116
Length of matches whith ID: 116 is : 100
Len of matches: 2600 at itteration 117
Length of matches whith ID: 117 is : 100
Len of matches: 2600 at itteration 118
Length of matches whith ID: 118 is : 100
Len of matches: 2600 at itteration 119
Length of matches whith ID: 119 is : 5
Len of matc

In [13]:
len(matches)

2609

In [23]:
5643292321 in matches

True

In [33]:
# a bit too high, this needs to be adjusted to not include event games

In [37]:
m = 5643292321

initialURL = 'https://api.steampowered.com/IDOTA2Match_570/GetMatchDetails/V001/'
match = pd.read_json(initialURL + '?match_id={}&key=0F68F5F96EC6CC04C40F0A28024E20C0'.format(m))
vals = pd.DataFrame(match.loc['players']['result']).set_index('account_id').loc[96887717]

In [32]:
match.loc['duration']['result']

2307

In [50]:
short_games = 0
games_browsed = 0
max_games_to_browse = 500
short_games_set = set()
for m in matches:
    initialURL = 'https://api.steampowered.com/IDOTA2Match_570/GetMatchDetails/V001/'
    match = pd.read_json(initialURL + '?match_id={}&key=0F68F5F96EC6CC04C40F0A28024E20C0'.format(m))
    # vals = pd.DataFrame(match.loc['players']['result']).set_index('account_id').loc[96887717]
    if match.loc['duration']['result'] < 600:
        short_games += 1
        print('Short_game', end = '.')
        short_games_set.add(m)
    games_browsed += 1
    if games_browsed > max_games_to_browse:
        break
    print('', end = '.')
    

..........................................................................................Short_game................................................................................................Short_game...................................................................................................................................................................................................................................................................................Short_game..Short_game.........................................

In [51]:
short_games

4

#### After looking in every single game, to see if the extra 200 games I shouldn't be seeing were 'abandoned' games (games that never) finished, I only found 18 'abandoned' games. So I still not sure where these extra games are comming from

At the time of wriing this:
- The API is returning **2,609** games.
- Other third party sites say I have **2,367** games and another says **2,388**
- The dotain-game client says I have **2,558**

### Moral?
- The data is messy and confusing (as usual)
- I will need to do more cleaning when I parse all the match data from every single game (before I was just getting match ID's and durations

In [47]:
short_games

2

In [48]:
short_games_set

{1932157245, 4955628114}